cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1431,7 +1431,9 @@ def _add_compute_resource_launch_template(
                 else None,  # parameter not supported for instance types with multiple network interfaces
                 interface_type="efa" if compute_resource.efa and compute_resource.efa.enabled else None,
                 groups=queue_lt_security_groups,
-                subnet_id=queue.networking.subnet_ids[0],
+                subnet_id=queue.networking.subnet_ids[0]
```

Comment:
I was thinking to always remove the subnet_id from the LT, so that the behavior is aligned with both launch API, i.e. the subnet info is retrieved form the fleet config interface. In this way the logic is more "linear" and there is no need for an if in here

Reply:
That makes sense, making the Node changes in a separate PR: [Use the SubnetId in the fleet-config for RunInstances-based launches #459](https://github.com/aws/aws-parallelcluster-node/pull/459)

Reply:
There is a limitation on specifying a SubnetID as part of RunInstances parameters when using a LaunchTemplate that already includes a NetworkInterface
```
"errorCode": "Client.InvalidParameterCombination",
"errorMessage": "Network interfaces and an instance-level subnet ID may not be specified on the same request"
```